Realm/RLMApp.h

```diff
@@ -0,0 +1,97 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import "RLMNetworkTransporting.h"
+
+#ifndef RLMApp_h
+#define RLMApp_h
+
+@class RLMSyncUser, RLMAppCredentials;
+
+/// A block type used for APIs which asynchronously vend an `RLMSyncUser`.
+typedef void(^RLMUserCompletionBlock)(RLMSyncUser * _Nullable, NSError * _Nullable);
+
+#pragma mark RLMAppConfiguration
+
+/// Properties representing the configuration of a client
+/// that communicate with a particular Realm application.
+@interface RLMAppConfiguration : NSObject
+
+/// A custom base URL to request against.
+@property (nonatomic, strong) NSString* _Nullable baseURL;
```

Comment:
Nullable properties should be marked with `nullable` in the property attributes list rather than annotating the pointer.